vertex-workbench-template/docs/network.md

```diff
@@ -22,10 +22,10 @@ All outgoing traffic is evaluated against VPC-wide [firewall rules](https://clou
-| `egress-allow-restricted-gcp-services` | egress    | 65543             | `199.36.153.4/30` | all                                                                                                                                                                                                                    | TODO: this can probably be removed, assuming we use user managed instances in favour of gcp managed instances. Need to test this before removing.                                                                                           |
-
+| `egress-allow-tcp-git`                 | egress    | 65534             | `140.82.112.0/20` | `80, 443`                                                                                                                                                                                                              | Corresponding rule to `ingress-allow-tcp-git`.                                                                                                                                                                                              |
+| `egress-allow-private-gcp-services`    | egress    | 65534             | `199.36.153.8/30` | all | Allow egress from instances on this network to the [IP range for `private.googleapis.com`](https://cloud.google.com/vpc/docs/configure-private-google-access-hybrid), which is only routable from within Google Cloud. |
+| `egress-allow-restricted-gcp-services` | egress    | 65534             | `199.36.153.4/30` | all                                                                                                                                                                                                                    | TODO: this can probably be removed, assuming we use user managed instances in favour of gcp managed instances. Need to test this before removing.                                                                                           |
+| `egress-allow-pypi-fastly`             | egress    | 65534             | `151.101.64.223, 199.232.36.223`  | `443`                                                                                                                                                                                                                  | Allow egress from instances in this network to PyPI and Fastly IPs. At the time of writing, these are the specific IPs resolved for the closest CDN for PyPI and Fastly from the Montreal datacenter, where the notebook instances run. We have to periodically review and update these IP addresses, as they may change due to infrastructure updates or CDN provider modifications.  <br> TODO: Set up a custom Artifactory on GCP to serve as a proxy for PyPI.                                   |
```

Comment:
Can you cite where you got these public IP addresses for PyPi and Fastly? E.g. using `dig`, some kind of API, etc.

Reply:
I've added the citations for obtaining the IPs